EIPS/eip-1108.md

```diff
@@ -48,15 +84,15 @@ Following is a table with the current gas cost and new gas cost:
 | ------------- | --------- | -----------------------------  | ------------------- |
 | `ECADD`       | `0x06`    | 500<sup>[1]</sup>              | 150                 |
 | `ECMUL`       | `0x07`    | 40 000<sup>[1]</sup>           | 6 000               |
-| Pairing check | `0x08`    | 80 000 * k + 100 000<sup>[2]</sup>| 28 300 * k + 35 450    |
+| Pairing check | `0x08`    | 80 000 * k + 100 000<sup>[2]</sup>| 34 000 * k + 45 000    |
 
 The gas costs for `ECADD` and `ECMUL` are updates to the costs listed in
 EIP-196, while the gas costs for the pairing check are updates to the cost
 listed in EIP-197. Updated gas costs have been adjusted to the less performant 
-client which is Parity, according to benchmarks<sup>[3]</sup>.
+client which is Parity, according to benchmarks<sup>[3]</sup>. The updated gas costs are scaled relative to the `ecrecover` precompile.
```

Comment:
Can you explain how it is relative to ecrecover?

Reply:
I pushed a change that describes the `ecrecover` scaling. The `ecrecover` precompile took 116 microseconds to run in the benchmark. If we assume the `ecrecover` gas price of 3,000 gas is a fair price, we get a price of 25.86 gas per microsecond of an precompile algorithm's run-time.  

The pairing precompile took 3,037 microseconds to compute 1 pairing, and 14,663 microseconds to compute 10 pairings. From this, the pairing algorithm has a fixed 'base' run-time of 1,745 microseconds, plus 1,292 microseconds per pairing. We can split the run-time into 'fixed cost' and 'linear cost per pairing' components because of the structure of the algorithm.  

Then using a 'fair' price of 25.86 gas per microsecond, we get a gas formula of ~`35,000 * k + 45,000` gas, where `k` is the number of pairings being computed.  

n.b. of course, I am also making an explicit assumption that the relative ratio between the run-times of `ecrecover` and the pairing precompile are constant irrespective of the specs or architecture of the underlying machine. I think this is a reasonable assumption, as the arithmetic operations being performed are extremely similar (modular multiplications in 256 bit finite fields). However more benchmarks would be useful here.

Reply:
@zac-williamson @Shadowfiend I think it would be nice to include the above text in the EIP. Can you add it?

Reply:
I believe it's in there, right? @ https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1108.md#specification, ‘i.e. in the benchmark…’

Reply:
Some of it is, but this is a more comprehensive description.